set.mm

```diff
@@ -457457,6 +457498,109 @@ and the expression ( x e. A /\ x e. B ) ` .
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Double restricted existential uniqueness quantification
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    $d a b p $.  $d a b ph $.  $d b x $.
+    opreu2reub.a $e |- ( p = <. a , b >. -> ( ph <-> ch ) ) $.
+    $( Lemma for ~ opreu2reub and other similar theorems.  (Contributed by
+       Thierry Arnoux, 4-Jul-2023.) $)
+    2reulem $p |- ( p = <. x , y >. -> ( ph <-> [. y / b ]. [. x / a ]. ch ) )
+       $=
+      ( cv cop wceq wsbc wb wi cvv wcel sbcth sbcim1 csb bitrd csbopg csbconstg
+      syl sbceq2g csbvarg opeq12d eqtrd eqeq2d sbcbig sbcg bibi1d 3imtr3d elv )
+      EIZCIZDIZJZKZABFUOLZGUPLZMZNDUPOPZUNUOGIZJZKZGUPLZAUSMZGUPLZURVAVBVEVGNZG
+      UPLVFVHNVIGUPOVICUOOPZUNFIZVCJZKZFUOLZABMZFUOLZVEVGVJVMVONZFUOLVNVPNVQFUO
+      OHQVMVOFUORUCVJVNUNFUOVLSZKVEFUOUNVLOUDVJVRVDUNVJVRFUOVKSZFUOVCSZJVDFUOVK
+      VCOUAVJVSUOVTVCFUOOUEFUOVCOUBUFUGUHTVJVPAFUOLZUSMVGABFUOOUIVJWAAUSAFUOOUJ
+      UKTULUMQVEVGGUPRUCVBVFUNGUPVDSZKURGUPUNVDOUDVBWBUQUNVBWBGUPUOSZGUPVCSZJUQ
+      GUPUOVCOUAVBWCUOWDUPGUPUOOUBGUPOUEUFUGUHTVBVHAGUPLZUTMVAAUSGUPOUIVBWEAUTA
+      GUPOUJUKTULUM $.
+
+    $d A a b p x y $.  $d B a b p x y $.  $d ph x y $.  $d ch p x y $.
+    $( Version of ~ opreu2reu with one less hypothesis.  (Contributed by
+       Thierry Arnoux, 4-Jul-2023.) $)
+    opreu2reus $p |- ( E! p e. ( A X. B ) ph -> E! a e. A E! b e. B ch ) $=
+      ( vx vy cv wsbc 2reulem opreu2reu ) ABFIKLGJKLBIJCDEFGHABIJEFGHMN $.
+
+    $( Version of ~ opreu2reurex1 with one less hypothesis.  (Contributed by
+       Thierry Arnoux, 4-Jul-2023.) $)
+    opreu2reurex1s $p |- ( E! p e. ( A X. B ) ph
+      -> ( E! a e. A E. b e. B ch /\ E! b e. B E. a e. A ch ) ) $=
+      ( vx vy cv wsbc 2reulem opreu2reurex1 ) ABFIKLGJKLBIJCDEFGHABIJEFGHMN $.
+
+    $d a b ch p x y $.
+    $( Correspondence between uniqueness of ordered pairs and double restricted
+       existential uniqueness quantification.  (Contributed by Thierry Arnoux,
+       4-Jul-2023.) $)
+    opreu2reub $p |- ( E! p e. ( A X. B ) ph
+                  <-> ( E! a e. A E. b e. B ch /\ E! b e. B E. a e. A ch ) ) $=
+      ( vx vy wreu wrex wa cv wceq wi wral wcel nfv nfan cxp opreu2reurex1s cop
```

Comment:
This theorem is identical with my theorem ~opreu2reurex which I provided with my recent PR #3277. My proof is much shorter, using ~reu3op and ~2reu4. Maybe the proof of ~opreu2reub is interesting (based on much simpler theorems), so it may be kept as ALT version.

Reply:
I renamed this ALT for reference, and only kept the direction which was new, so it does not refer to ~ opreu2reurex1.